### DISCUSSION TITLE
Whitelisting certain websites

### DISCUSSION THREAD

**Discussion by Amir1453** [netblue30/firejail#5833] [Q&A] [answered]
Hello, I apologize for my ignorance, but I was unable to find a solution to what I am looking for, and my networking skills are incredibly poor. 

Is there any way to sandbox the application in such a way that it can only access certain websites? For example allowing only github to be accessed. 

Please let me know how this can be done, and if not any alternatives.

**Comment by rusty-snake** [comment, marked as answer]
You may want to have a look at [OpenSnitch](https://github.com/evilsocket/opensnitch) and [Portmaster](https://safing.io/).

Other Discussions: #4834, #5673

With firejail you have to options

### DNS based

You can use a program like [fdns](https://github.com/netblue30/fdns) to restrict the set of domains which can be resolved.

Advantages:
- Fine grained control over allowed and disallowed domains.

Disadvantage:
- Can be bypassed using
  - hardcoded IP addresses
  - seperate resolver library

=> Works if the goal is privacy (e.g. use "big-tech" program in restricted environment) but not if the goal is security.

### IP based

You can use `netfilter` with a custom netfilter that only allows a restricted set of `proto/ip:port`.

Advantages:
- Can not be bypassed <sub>at least not easily and not without any kind of sandbox escape</sub>

Disadvantage:
- `proto/ip:port` must not map to a specific domain/server/service. Especially for servers hosted behind a CDN, the number of servers behind a single IP address can be huge.
- firejail's `net` command has some conveniences drawbacks.

### `--netlock`

There's also `--netlock` feature.

**Comment by Amir1453** [comment]
Firstly, thank you for taking the time for such a detail answer. Just to clarify, I will be using this with Obsidian, a markdown text editor. I guess I will be using fdns.

Quick question about the --netlock feature, how long is the recorded traffic kept? Does it renew every time you start the application? Is there a way to see the traffic and perhaps edit if necessary?

**Reply by rusty-snake** [reply]
--netlock store only in memory I.e. it is recorded at every start.

**Reply by Stoxis** [reply]
> Firstly, thank you for taking the time for such a detail answer. Just to clarify, I will be using this with Obsidian, a markdown text editor. I guess I will be using fdns.
> 
> Quick question about the --netlock feature, how long is the recorded traffic kept? Does it renew every time you start the application? Is there a way to see the traffic and perhaps edit if necessary?

If you figure this out please share your profile with me, I want a profile for Obsidian but my network skills are very sub-par and I've never really used this firejail app before.

**Reply by Amir1453** [reply]
I will create a repository about it soon. I will link it here when it is finished.

**Reply by rusty-snake** [reply]
If you follow the profile template (which contains best practices and helpful tips) you can open a PR here so that anyone benefits of it with the next release.

**Reply by Amir1453** [reply]
Thanks for the suggestion, I will create a pull request after I finish the GitHub repository and after rechecking the profile.

**Reply by Amir1453** [reply]
@Stoxis Here is the GitHub repository: https://github.com/Amir1453/obsidicrypt
